src/library/scala/collection/Map.scala

```diff
@@ -281,12 +282,8 @@ trait MapOps[K, +V, +CC[_, _] <: IterableOps[_, AnyConstr, _], +C]
   /** Alias for `concat` */
   /*@`inline` final*/ def ++ [V2 >: V](xs: collection.Iterable[(K, V2)]): CC[K, V2] = concat(xs)
 
-  override def mkString(start: String, sep: String, end: String): String =
-    iterator.map { case (k, v) => s"$k -> $v" }.mkString(start, sep, end)
-
-  // these dummy overrides are necessary for disambiguation
-  override def mkString(sep: String): String = super.mkString(sep)
-  override def mkString: String = super.mkString
+  override def addString(sb: StringBuilder, start: String, sep: String, end: String): StringBuilder =
+    iterator.map { case (k, v) => s"$k -> $v" }.addString(sb, start, sep, end)
```

Comment:
This looks sillier now that the `s""` uses a `StringBuilder`. Maybe it's worth writing this out here. Or is it worth adding `addElementString` for `Map` to override?